content/Cookies/Cookies.q.md

```diff
@@ -0,0 +1,947 @@
+
+if (isset($_COOKIE['user'])) {
+	echo "The cookie value is: " . $_COOKIE'user'];
+}
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is the error message?
+
+ - Fatal error: Cannot use `[]` for reading on line number 15
+
+ - Warning: Cookie names must not be empty on line number 6
+
+ - syntax error, unexpected `'60'` (T_LNUMBER) on line number 4
+
+ - syntax error, unexpected `'time'` (T_STRING), expecting `','` or `')'` on line number 2
+
+ - syntax error, unexpected `''user''` (T_CONSTANT_ENCAPSED_STRING), expecting `','` or `';'` on line number 5
+
+
+/// type=MS, answer=[1,2]
+
+Which statements correctly describe the error?
+
+ - There is no open square bracket `[` after `$_COOKIE` on line 5.
+
+ - On line 5, `$_COOKIE'user']` in the `echo` statement is invalid.
+
+ - There is no semicolon `;` at the end of the statement on line 2.
+
+ - There is no addition operator `+` specified between `time()` and `60` on line 2.
+
+ - There is no open curly brace `{` after `if (isset($_COOKIE['user']))` on line 4.
+
+:::
+
+
+/// type=CR, answer=[tests/Cookies/MissingSquareBracketTest.php]
+
+Correct the code so that it outputs the string `The cookie value is: My Cookie`.
+
+```php
+<?php
+setcookie("user", "My Cookie", time() + 60);
+
+if (isset($_COOKIE['user'])) {
+	echo "The cookie value is: " . $_COOKIE'user'];
+}
+```
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+$cookieName = "my_cookie";  
+$cookieValue = "This is my cookie.";
+$expiry = time()  60;
+
+setcookie($cookieName, $cookieValue, $expiry);  
+?>
+
+<html>
+	<head>
+		<title>Creating and Accessing Cookie</title>
+	</head>
+	<body>  
+		<?php  
+		if(!isset($_COOKIE[$cookieName])) {  
+			echo "Sorry, a cookie is not found!";  
+		} else {  
+			echo "<br/>Cookie Value: " . $_COOKIE[$cookieName];  
+		}  
+		?>  
+	</body>  
+</html>
+```
+/// type=SS, answer=[3]
+
+Execute the program. What is the error message?
+
+ - Fatal error: Cannot use `[]` for reading on line number 15
+
+ - Warning: Cookie names must not be empty on line number 6
+
+ - syntax error, unexpected `'60'` (T_LNUMBER) on line number 4
+
+ - syntax error, unexpected `'time'` (T_STRING), expecting `','` or `')'` on line number 2
+
+ - syntax error, unexpected `''user''` (T_CONSTANT_ENCAPSED_STRING), expecting `','` or `';'` on line number 5
+
+
+/// type=MS, answer=[1,3]
+
+Which statements correctly describe the error?
+
+ - On line 4, the statement `$expiry = time()  60;` is invalid.
+
+ - There is no semicolon `;` at the end of the statement on line 3.
+
+ - There is no addition operator `+` specified between `time()` and `60` on line 4.
+
+ - On line 6, the argument `$cookieValue` is not enclosed in double quotes `""`.
+
+ - There is no comma `,` specified between `$cookieValue` and `$expiry` on line 6.
+
+:::
+
+
+/// type=CR, answer=[tests/Cookies/MissingPlusSignOnExpiryTest.php]
+
+Correct the code so that it outputs the string `Cookie Value: This is my cookie. `.
```

Comment:
Similar to my comment on line 589.

"Correct the code so that it outputs the string `Sorry, a cookie is not found!` on its first execution and the string `The cookie value is: My cookie` after it is reloaded"

Reply:
I don't think we still mention this statement since this will only occur if there is no cookie being created. I think I need to specify that it should create a cookie first. Thank you.